src/nanorc/tui.py

```diff
@@ -0,0 +1,589 @@
+                Button("Execute Command", id="go"),
+                Button('Cancel',variant='error', id='cancel'),
+                classes = "horizontalbuttonscontainer"
+            ), 
+            Static(id='errordisplay')
+        )
+    
+    async def on_button_pressed(self, event: Button.Pressed) -> None:
+        button_id = event.button.id
+        params_out = {}
+        inputs = self.query(Input)
+        #A list of all statics with the given id. There should only be one of these, so we get element 0
+        errordisplay = [q for q in self.query(Static) if q.id == "errordisplay"][0]
+        for s in self.app.query(Static):    #This gets all the children of the app, since they all inherit from static
+                if isinstance(s, Status):
+                    status_obj = s
+                if isinstance(s, Command):
+                    command_obj = s
+
+        if button_id == "go":
+            for i in inputs:
+                if i.value == "":     
+                    if self.params[i.id]['required']:       #We can't allow a required field to be empty!
+                        errordisplay.update(f"Required parameter \"{i.id}\" is missing!")
+                        return                              #An attempt at sending a command is over
+                else:
+                    #We enforce here that the string will be convertable to the appropriate type
+                    match self.params[i.id]['type']:
+                        case "STRING":
+                            pass        #This is a valid input type, so it's included in case we need to do something later
+                        case "INT":
+                            if not i.value.isdigit():
+                                errordisplay.update(f"{i.value} is not a valid input for \"{i.id}\". Input should be an integer.")
+                                return
+                        case "FLOAT":
+                            try:
+                                f = float(i.value)
+                            except:
+                                errordisplay.update(f"{i.value} is not a valid input for \"{i.id}\". Input should be a float.")
+                                return
+                        case "BOOL":
+                            if (i.value.lower() != "true") and (i.value.lower() != "false"):
+                                errordisplay.update(f"{i.value} is not a valid input for \"{i.id}\". Input should be a boolean.")
+                                return
+                    params_out[i.id] = i.value
+
+            payload = {'command': self.command, **params_out}
+            if 'timeout' in payload:
+                t = int(payload['timeout'])
+            else:
+                t = 180
+
+            async with httpx.AsyncClient() as client:
+                r = await client.post(f'{self.hostname}/nanorcrest/command', auth=("fooUsr", "barPass"), data=payload, timeout=t)
+
+            if "logs" in r.json():
```

Comment:
I've put the output of r.json() into a variable and used that instead of potentially converting from string to dict over and over. (this was done for other parts of the code too)